src/gui/clips/ClipView.cpp

```diff
@@ -557,8 +557,11 @@ DataFile ClipView::createClipDataFiles(
 	dataFile.content().appendChild( clipParent );
 
 	// Add extra metadata needed for calculations later
-	int initialTrackIndex = tc->tracks().indexOf( t );
-	if( initialTrackIndex < 0 )
+
+	const auto initialTrackIt = std::find(tc->tracks().begin(), tc->tracks().end(), t);
+	const int initialTrackIndex = initialTrackIt != tc->tracks().end() ? std::distance(tc->tracks().begin(), initialTrackIt) : -1;
+
+	if (initialTrackIndex < 0)
```

Comment:
How about checking `initialTrackIt != tc->tracks().end()` in `if` and then define `initialTrackIndex` after it?

Reply:
That would make the most sense since the ``if`` is already meant to check if it is not found.. :man_facepalming:. Thanks.